sdk/iot/hub/samples/src/generate_certificate.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+echo "Generating DEMO User Root Cert"
+echo "DO NOT USE THIS IN PRODUCTION"
+openssl ecparam -out user_root_cert_key.pem -name prime256v1 -genkey
+openssl req -new -days 365 -nodes -x509 -key user_root_cert_key.pem -out user_root_cert.pem -subj "/CN=CA Group 1"
+
+echo "Generating DEMO User Device Cert"
+echo "DO NOT USE THIS IN PRODUCTION"
+openssl ecparam -out device_ec_key.pem -name prime256v1 -genkey
```

Comment:
This should be a self-signed certificate with the proper note that it should not be used in production and that device certificates should be generated using a hardware-based random-number-generator.

Reply:
do we need generate the whole hierarchy (CA -> leaf cert)?

It should be simple to use single self signed cert and use that?  Both DPS and IOT hub accepts that .



---
In reply to: [420333718](https://github.com/Azure/azure-sdk-for-c/pull/682#discussion_r420333718) [](ancestors = 420333718)